UI/users/preferences.html

```diff
@@ -80,17 +82,18 @@
                                   default_values = [ user.prefs.numberformat ],
                                   text_attr = 'format',
                                   value_attr = 'format',
+                                  required = 1
                                 } %]</td>
                           </tr>
                           <tr>
                             <th align="right"><label for="language">[% text('Language') %]</label></th>
-                            <td>[% PROCESS select_language element_data={
+                            <td>[% PROCESS select element_data={
                                 name = 'language',
                                 options = user.language_codes,
                                 default_values = [ user.prefs.language ],
                                 text_attr = 'label',
                                 value_attr = 'id',
-                                default_blank = 1,
+                                required = 1
```

Comment:
Why is it required? When the user doesn't choose anything, the system default will be selected. Isn't that just fine?

Reply:
System default can be anything. We support about 40 over the 200+ possible choices

Reply:
Yes. It can be anything. I think that was my point before when I told you that we only have 40 languages and that we now offer a list of 200 options. So I'm not sure what impact either has on the point that I tried to make that very often a system wide configured language (in ledgersmb.conf) is likely to fit at least 80% of the users. Or that the system wide default (`en`) is going to suit a lot of users in our target market in general. Anyway: I don't think that setting the first language in the list is a good default when there's no configured language in the user's settings -- which is what the above achieves.

Reply:
1.8 was offering 200+ languages. 1.9 limits them to the languages listed in the database and for that, it should be required and not defaulted to whatever locale may be used on the user system or manually put in ledgersmb.conf.

Reply:
Ok. But on my test system, I make new users all the time. These users use the hardcoded default language "en" because they are created from setup.pl. In the preference page, the first language in the list is selected. If I submit that page without explicitly selecting a language, the language setting changes to Basque. I was happy with the system default, but I'm not happy with Basque. What to do about this?

Reply:
Git is _not_ showing it in this conversation but _default_blank_ is restored since your first comment.

Reply:
Sorry I didn't notice that! Done hint on part of GitHub would have been practical!